tree/master/cloud/src/solution/Microsoft.Legal.MatterCenter.Service/Microsoft.Legal.MatterCenter.Repository/Configs/ConfigRepository.cs

```diff
@@ -21,16 +23,68 @@ public class ConfigRepository : IConfigRepository
         private UIConfigSettings configSettings;
         private IConfigRepository config;
 
-
+        /// <summary>
+        /// ConfigRepository with all the required dependency injections inserted
+        /// </summary>
+        /// <param name="generalSettings"></param>
+        /// <param name="configSettings"></param>
         public ConfigRepository(
             IOptions<GeneralSettings> generalSettings,
               IOptions<UIConfigSettings> configSettings)
         {
             this.generalSettings = generalSettings.Value;
             this.configSettings = configSettings.Value;
+            
         }
 
 
+        /// <summary>
+        /// This method will return the secure password for authentication to SharePoint Online
+        /// </summary>
+        /// <param name="plainTextPassword"></param>
+        /// <returns></returns>
+        public static SecureString GetEncryptedPassword(string plainTextPassword)
+        {
+            //Get the user's password as a SecureString
+            SecureString securePassword = new SecureString();
+            foreach (char c in plainTextPassword)
+            {
+                securePassword.AppendChar(c);
+            }
+            //while (info.Key != ConsoleKey.Enter);
+            return securePassword;
+        }
+
+        /// <summary>
+        /// This method will upload uiconfigforspo.js into sharepoint catalog site collection
+        /// </summary>
+        /// <param name="filePath"></param>
+        /// <param name="clientUrl"></param>
+        public void UploadConfigFileToSPO(string filePath, string clientUrl)

```

Comment:
new method to upload ui config for sharepoint online. Once that file is uploaded to sharepoint online, ui config for spo js file will get deleted from azure web site